### ISSUE TITLE
IEI-49757 Compression of frames on storage results in Compressor$CompressedFrame on finalizer queue

### ISSUE THREAD

**Issue by mfarooq2000** [dcm4che/dcm4che#181]
Heap dumps showing the Finalizer queue had items on it from the dcm4che3 Compressor:
Compressor$CompressedFrame has:

                #cache = new MemoryCacheImageOutputStream(cacheout) {

                    @Override
                    public void flush() throws IOException {
                        // defer flush to writeTo()
                        LOG.debug("Ignore invoke of MemoryCacheImageOutputStream.flush()");
                    }
                };
The MemoryCacheImageOutputStream class extends ImageOutputStreamImpl, which extends ImageInputStreamImpl, which has a finalize() method declared.
These objects being on the Finalizer queue can result in excessive memory usage by Compressor objects that hang around until (potentially) being cleaned up by [GC.[](url)](url)

**Comment by hczedik** [comment]
Are you planning to also open this as a pull request for the master branch?

**Comment by mfarooq2000** [comment]
Yes, I will verify the changes on master branch and create a pull request.
